airflow/settings.py

```diff
@@ -183,6 +183,22 @@ def configure_orm(disable_connection_pool=False):
     # For Python2 we get back a newstr and need a str
     engine_args['encoding'] = engine_args['encoding'].__str__()
 
+    # Sane timeouts for connections and statements
+    connect_args = {}
+
+    try:
+        connect_timeout_seconds = conf.getint('core', 'SQL_ALCHEMY_CONNECT_TIMEOUT_SECONDS')
+    except conf.AirflowConfigException:
+        connect_timeout_seconds = 30
+    connect_args['connect_timeout'] = connect_timeout_seconds
+
+    try:
+        statement_timeout_seconds = conf.getint('core', 'SQL_ALCHEMY_STATEMENT_TIMEOUT_SECONDS')
+    except conf.AirflowConfigException:
+        statement_timeout_seconds = 30
+    connect_args['options'] = '-c statement_timeout={}'.format(statement_timeout_seconds * 1000)
```

Comment:
This only works when Postgres is the backend. MySQL has a different thing iirc, and SQLite doesn't really have connections. Can just gate this to Postgres only for now probably, with a note that support should be added for MySQL